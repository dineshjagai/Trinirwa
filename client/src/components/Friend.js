import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import BlockIcon from '@material-ui/icons/Block';
import ChatIcon from '@material-ui/icons/Chat';
import Tooltip from '@material-ui/core/Tooltip';
import PersonAddDisabledIcon from '@material-ui/icons/PersonAddDisabled';
import './Follower.css';

const useStyles = makeStyles((theme) => ({
  root: {
    '& > *': {
      margin: theme.spacing(1.5),
      borderRadius: '100%',
      padding: 0,
      overflow: 'none',
    },
  },
}));

export default function Friend({
  info, handleB, handleMessage, handleU,
}) {
  const classes = useStyles();
  return (
    <div className="Container_two">
      <div className={classes.root}>
        <Tooltip title="Message" placement="top">
          <ChatIcon className="button" onClick={() => handleMessage} />
        </Tooltip>
        <Tooltip title="Block user" placement="top">
          <BlockIcon className="button" color="secondary" onClick={() => handleB(info.username)} />
        </Tooltip>
        <Tooltip title="Unfollow" placement="top">
          <PersonAddDisabledIcon className="button" onClick={() => handleU(info.username)} />
        </Tooltip>
      </div>
      <div className="followerContainer">
        <div className="myimgContainer">
          <img
            className="myimg"
            src={`/viewFile/${info.profile_picture}`}
            alt=""
          />
        </div>
      </div>
      <div className="text">{info.username}</div>
    </div>
  );
}
