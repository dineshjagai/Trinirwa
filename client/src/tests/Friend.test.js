import React from 'react';
import renderer from 'react-test-renderer';
import Friend from '../components/Friend';

test('Test Friend snapshot', () => {
  const f = {
    'username' : 'isimbib',
    'profile_picture' :  'logo512.png',
  };
  const component = renderer.create(<Friend info={f}/>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
