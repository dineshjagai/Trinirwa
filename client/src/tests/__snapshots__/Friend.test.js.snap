// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Friend snapshot 1`] = `
<div
  className="Container_two"
>
  <div
    className="makeStyles-root-1"
  >
    <svg
      aria-describedby={null}
      aria-hidden={true}
      className="MuiSvgIcon-root button"
      focusable="false"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      title="Message"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zM6 9h12v2H6V9zm8 5H6v-2h8v2zm4-6H6V6h12v2z"
      />
    </svg>
    <svg
      aria-describedby={null}
      aria-hidden={true}
      className="MuiSvgIcon-root button MuiSvgIcon-colorSecondary"
      focusable="false"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      title="Block user"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM4 12c0-4.42 3.58-8 8-8 1.85 0 3.55.63 4.9 1.69L5.69 16.9C4.63 15.55 4 13.85 4 12zm8 8c-1.85 0-3.55-.63-4.9-1.69L18.31 7.1C19.37 8.45 20 10.15 20 12c0 4.42-3.58 8-8 8z"
      />
    </svg>
    <svg
      aria-describedby={null}
      aria-hidden={true}
      className="MuiSvgIcon-root button"
      focusable="false"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      title="Unfollow"
      viewBox="0 0 24 24"
    >
      <circle
        cx="15"
        cy="8"
        r="4"
      />
      <path
        d="M23 20v-2c0-2.3-4.1-3.7-6.9-3.9l6 5.9h.9zm-11.6-5.5C9.2 15.1 7 16.3 7 18v2h9.9l4 4 1.3-1.3-21-20.9L0 3.1l4 4V10H1v2h3v3h2v-3h2.9l2.5 2.5zM6 10v-.9l.9.9H6z"
      />
    </svg>
  </div>
  <div
    className="followerContainer"
  >
    <div
      className="myimgContainer"
    >
      <img
        alt=""
        className="myimg"
        src="/api/viewFile/logo512.png"
      />
    </div>
  </div>
  <div
    className="text"
  >
    isimbib
  </div>
</div>
`;
